### PR TITLE
Order functionality applied

### PR DESCRIPTION
**Pull request by QbaSekowski**

### CODE REVIEW

File: src/main/java/mate/academy/springbookapp/service/order/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemDto> getAllItemsFromOrder(Long userId, Long orderId) {
+        Order modelOrder = orderRepository.findByIdWithItems(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with id: " + orderId));
+        if (!modelOrder.getUser().getId().equals(userId)) {
+            return null;
+        }
```
Comment (nowickimj) [inline src/main/java/mate/academy/springbookapp/service/order/impl/OrderServiceImpl.java:47]: 1. Avoid using `null` when possible.
2. Think of how should your application behave when user tries to access someone else's resources. I think that you should either return 404 (more secure - you're not telling anyone that resource exists, but may be misleading for debugging) or 403 (more explicit).

File: src/main/java/mate/academy/springbookapp/service/order/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        OrderItem modelItem = modelOrder.getOrderItems().stream()
+                .filter(oi -> oi.getId().equals(itemId))
+                .findFirst()
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Can't find item with id: " + itemId));
+        return orderItemMapper.toDto(modelItem);
+    }
+
+    @Override
+    public OrderDto placeOrder(User user, PlaceOrderRequestDto requestDto) {
```
Comment (nowickimj) [inline src/main/java/mate/academy/springbookapp/service/order/impl/OrderServiceImpl.java:69]: How about cleaning up this method a bit? First - prepare data, then create new `Order` object and set values. Additionally, you might want to try out Lombok's `@Builder`.

File: src/main/java/mate/academy/springbookapp/dto/order/UpdateStatusRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package mate.academy.springbookapp.dto.order;
+
+import jakarta.validation.constraints.NotBlank;
+
+public record UpdateStatusRequestDto(
+        @NotBlank
+        String status) {
```
Comment (nowickimj) [inline src/main/java/mate/academy/springbookapp/dto/order/UpdateStatusRequestDto.java:7]: How about using enum here?

**Review by nowickimj — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]
A few suggestions:
Don't use @ Data on JPA entities, it has [side effects](https://thorben-janssen.com/lombok-hibernate-how-to-avoid-common-pitfalls/#avoid-data)
Don't use short uninformative variables' names in stream chains
Don't forget to leave the last line empty (regarding your .yaml scripts)
Extract /api to project wide prefix - `server.servlet.context-path`
@ Where is deprecated, replace with @ SQLRestriction

**Pull request merged**